### PR TITLE
Get registers

### PR DESCRIPTION
**Pull request by sc108-lee**

### CODE REVIEW

File: src/nvme/linux.c [hunk 1]
```diff
@@ -494,6 +496,139 @@ char *nvme_get_path_attr(nvme_path_t p, const char *attr)
...
+
+	for (i = 0; i < num_ctrls; i++) {
+		if (strncmp(dev, ctrls[i]->d_name, strlen(ctrls[i]->d_name)) == 0) {
+			base = ctrls[i]->d_name;
+			break;
+		}
+	}
+
+	if (!base) {
+		fprintf(stderr, "Can't find %s in scaned ctrls\n", dev);
```
Comment (martin-belanger) [inline src/nvme/linux.c:588]: minor typo: `scaned` -> `scanned`

File: src/nvme/linux.c [hunk 2]
```diff
@@ -494,6 +496,139 @@ char *nvme_get_path_attr(nvme_path_t p, const char *attr)
...
+		goto free_ctrls;
+	}
+
+	sprintf(path, "/sys/class/nvme/%s/transport", base);
```
Comment (martin-belanger) [inline src/nvme/linux.c:592]: In `libnvme`, temporary strings are usually dynamically allocated with `asprintf()`. Obviously, this also requires a `free()` before exiting.

File: src/nvme/linux.h [hunk 3]
```diff
@@ -176,6 +176,25 @@ int nvme_namespace_detach_ctrls(int fd, __u32 nsid, __u16 num_ctrls, __u16 *ctrl
  */
 int nvme_open(const char *name);
 
+/**
+ * nvme_get_registers() - Read registers mmap or with fabrics command
+ * @fd:			Opended target device file descriptor
```
Comment (martin-belanger) [inline src/nvme/linux.h:181]: typo: `Opended` -> `Opened`

File: src/nvme/linux.c (hunk 1 above)
Comment (sc108-lee) [reply src/nvme/linux.c]: thanks for checking fixed :)

File: src/nvme/linux.c (hunk 2 above)
Comment (sc108-lee) [reply src/nvme/linux.c]: thanks for checking fixed :)

File: src/nvme/linux.h (hunk 3 above)
Comment (sc108-lee) [reply src/nvme/linux.h]: thanks for checking fixed :)

**Comment by igaw** [comment]
This PR is part of https://github.com/linux-nvme/nvme-cli/pull/1541

The current head doesn't issue any ioctl anymore when scanning the topology. Thus we don't need add this here anymore.

**Pull request closed without merging**